File: server/src/main/java/com/example/schoolsite/services/UserService.java
package com.example.schoolsite.services;

import com.example.schoolsite.dto.UserDTO;
import com.example.schoolsite.entity.User;
import com.example.schoolsite.exception.ResourceNotFoundException;

import java.util.List;

public interface UserService {

    List<User> getAllUsers();

   UserDTO getUserById(Long userId)
            throws ResourceNotFoundException;

    UserDTO updateUser(Long userId,
                           UserDTO userDetails) throws ResourceNotFoundException;
}

File: server/src/main/java/com/example/schoolsite/services/UserService.kt
package com.example.schoolsite.services

import com.example.schoolsite.dto.UserDTO

interface UserService {
    val allUsers: List<Any?>?

    @Throws(ResourceNotFoundException::class)
    fun getUserById(userId: Long?): UserDTO?

    @Throws(ResourceNotFoundException::class)
    fun updateUser(userId: Long?,
                   userDetails: UserDTO?): UserDTO?
}